### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by hyoong**
Added references, hyperlink and hyperref in the latex/pdf. Added grover's algorithm circuit in section 2. Proof read my texts and made adjustments to the best of my ability.

### CODE REVIEW

File: docs/report/main.tex [hunk 1]
```diff
@@ -23,9 +23,21 @@
...
+\usepackage[superscript,biblabel]{cite}
 \usepackage[scale=1,bmargin=1.5cm,footnotesep=1.5cm]{geometry}
 
+\usepackage{hyperref}
+\hypersetup{
+    colorlinks=true,
+    linkcolor=violet,
+    filecolor=green,      
+    urlcolor=blue,
+    pdftitle={Overleaf Example},
```
Comment (Tiernan8r) [inline docs/report/main.tex:35]: Should this be modified to "Quantum Computing Project" or something along those lines?

File: docs/report/main.tex [hunk 2]
```diff
@@ -127,31 +140,30 @@ \subsection{Qubits}
...
 \vspace{5mm}
 
 \noindent
-The qubits in a quantum computer is stored in a quantum register. The size of the quantum register is dictated by the number of qubits. 
+The qubits in a quantum computer are stored in a quantum register. The size of the quantum register is dictated by the number of qubits. 
 \vspace{5mm}
 
 \noindent
-The vector state of the quantum register $\mathcal{H}$ with $n$ qubits can be expressed in tensor product:
+The vector state of the quantum register $\mathcal{H}$ with $n$ qubits can be expressed in tensor product\hyperlink{1}{$^1$}:
```
Comment (Tiernan8r) [inline docs/report/main.tex:162]: for citations, you should be able to use the `\cite{}` command, with a reference to the label for that citation, and latex will figure out the citation numbering when rendering the PDF

File: docs/report/main.tex [hunk 3]
```diff
@@ -825,12 +851,125 @@ \subsection{How The Algorithm Works}
...
+% Text Node
+\draw (316,95) node [anchor=north west][inner sep=0.75pt]   [align=left] {Grover's Diffusion Operator};
+% Text Node
+\draw (313,213) node [anchor=north west][inner sep=0.75pt]   [align=left] {Repeat $\displaystyle \approx \ \frac{\pi }{4}\sqrt{N}$ times};
+
+
+\end{tikzpicture}
 \vspace{5mm}
 
+\textbf{Quantum circuit representation of Grover's Algorithm.}
```
Comment (Tiernan8r) [inline docs/report/main.tex:968]: For a follow up, but we should figure out how to get LaTeX to auto number our figures.

File: docs/report/main.tex [hunk 4]
```diff
@@ -1057,17 +1198,43 @@ \subsection{Algorithmic issues}
 %	PROBLEM 2
 %----------------------------------------------------------------------------------------
 
-\section{References and Bibliographies}
 
+\section{References and Bibliographies}
```
Comment (Tiernan8r) [inline docs/report/main.tex:1202]: We should probably convert this to a BibTex citation in a separate file, and have LaTeX auto populate the bibliography.

**Review by Tiernan8r — APPROVED** [review]
Apart from a few musings from me, looks good!

File: docs/report/main.tex (hunk 1 above)
Comment (hyoong) [reply docs/report/main.tex:35]: Good shout, will change this

File: docs/report/main.tex (hunk 2 above)
Comment (hyoong) [reply docs/report/main.tex:162]: Yes, but ran into trouble with \cite due to conflict with the table of contents. Using \cite will create its own Reference header which is independent from the table of content numbering system. So you effective get two "References " one numbered and one not(due to \cite). So I came out with this slightly long winded method. Let me know if you can find a fix ☺️

File: docs/report/main.tex (hunk 3 above)
Comment (hyoong) [reply docs/report/main.tex:968]: Ok I will look into this .

File: docs/report/main.tex (hunk 4 above)
Comment (hyoong) [reply docs/report/main.tex:1202]: See comments on line 162

File: docs/report/main.tex (hunk 2 above)
Comment (Tiernan8r) [reply docs/report/main.tex:162]: https://github.com/Tiernan8r/quantum_computing_project/pull/78 changes to use cite. Have to make a bit of a workaround to remove the double section headering as you said.

**Pull request merged**